Pages/Index.cshtml.cs

```diff
@@ -1,47 +1,71 @@
-			       "+ dotnet new web --force\n" +
-			       "The template \"ASP.NET Core Empty\" was created successfully.\n" +
-			       "\n" +
-			       "Processing post-creation actions...\n" +
-			       "Running 'dotnet restore' on /mnt/ramdisk/dotnet-bunny/dotnet-regular-tests/createdump-aspnet/createdump-aspnet.csproj...\n" +
-			       "Restoring packages for /mnt/ramdisk/dotnet-bunny/dotnet-regular-tests/createdump-aspnet/createdump-aspnet.csproj...\n" +
-			       "Generating MSBuild file /mnt/ramdisk/dotnet-bunny/dotnet-regular-tests/createdump-aspnet/obj/createdump-aspnet.csproj.nuget.g.props.\n" +
-			       "    Generating MSBuild file /mnt/ramdisk/dotnet-bunny/dotnet-regular-tests/createdump-aspnet/obj/createdump-aspnet.csproj.nuget.g.targets.\n" +
-			       "    Restore completed in 1.22 sec for /mnt/ramdisk/dotnet-bunny/dotnet-regular-tests/createdump-aspnet/createdump-aspnet.csproj.\n" +
-			       "\n" +
-			       "Restore succeeded.\n" +
-			       "\n" +
-			       "+ dotnet tool install -g dotnet-dev-certs\n" +
-			       "Tool 'dotnet-dev-certs' is already installed.\n" +
-			       "+ true\n" +
-			       "+ dotnet dev-certs https\n" +
-			       "Cannot find command 'dotnet dev-certs', please run the following command to install\n" +
-			       "\n" +
-			       "dotnet tool install --global dotnet-dev-certs";
-
 		public List<BunnyJob> GetJobs()
 		{
-			//TODO...
+            var job_names=Directory.GetDirectories(JobsConfig.Get().JobsDirectory, JobsConfig.Get().JobsPrefix+"*");
+			var jobs=new List<BunnyJob>(job_names.Length);
+			var client=new WebClient();
 
-			return new List<BunnyJob>(){
-				new BunnyJob("dotnet-rpm-rhel7-x64-1.0", true),
-				new BunnyJob("dotnet-rpm-rhel7-x64-1.1", true),
-				new BunnyJob("dotnet-rpm-rhel7-x64-2.1", false, new List<BunnyTest>(){new BunnyTest("createdump-aspnet", GetExampleLog()), new BunnyTest("bash-completion", "Bash completion is bork and this is the example log.")}),
-				new BunnyJob("dotnet-rpm-rhel7-x64-2.2", true)
-			};
+			for(int i=0; i<job_names.Length; ++i)
+			{
+				job_names[i]=job_names[i].Substring(job_names[i].LastIndexOf('/')+1);
+				var build_names=Directory.GetDirectories(JobsConfig.Get().JobsDirectory+job_names[i]+"/builds");
+				int last_build=-1;
+				for(int j=0; j<build_names.Length; ++j)
+				{
+					int num=0;
+					if(int.TryParse(build_names[i].Substring(build_names[i].LastIndexOf('/')+1), out num))
+					{
+						last_build=last_build>num?last_build:num;
+					}
+				}
+				string logfile="";
+				try
+				{
+					logfile=client.DownloadString(JobsConfig.Get().JobsURL+job_names[i]+"/ws/results/logfile.log");
+				}
+				catch(WebException)
+				{
+					jobs.Add(new BunnyJob(job_names[i], last_build==-1?"":JobsConfig.Get().JobsURL+job_names[i]+"/"+last_build+"/console", false));
+					continue;
+				}
+				var errors=new List<int>(){0};
+				while(errors.Last()!=-1)
+				{
+					errors.Add(logfile.IndexOf("Result: FAIL", errors.Last()+1));

```

Comment:
This seems fairly inefficient, though it is probably fine for such a small use case.

Ultimately, this will have the walk the errors roughly N-factorial times (where N is the number of errors + 1). It would probably be better to get a list of all errors and then walk through them once.